### PR TITLE
Demo of auto prism language loading

### PR DESCRIPTION
**Pull request by Dhaulagiri**
### :link: External links

Jira ticket: [HDS-XXX](https://hashicorp.atlassian.net/browse/HDS-XXX)
Figma file: [if it applies]

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression
- [ ] A11y tests have been run locally (`yarn test:a11y --filter="COMPONENT-NAME"`)
- [ ] If documenting a new component, an acceptance test that includes the `a11yAudit` has been added
- [ ] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: packages/components/index.js [hunk 1]
```diff
@@ -12,8 +12,14 @@ module.exports = {
     return true;
   },
 
-  included: function (/* app */) {
+  included: function (app) {
     this._super.included.apply(this, arguments);
+    if (app.options && app.options['hds-code-block']) {
+      const options = app.options['hds-code-block'];
```
Comment (Dhaulagiri) [inline packages/components/index.js:18]: [Example](https://github.com/froala/ember-froala-editor/blob/19fee2c9a2a0ec3a475a504773eb5ac88738fe6f/index.js#L123-L128) of how we could even validate that these languages exist

File: packages/components/index.js [hunk 2]
```diff
@@ -12,8 +12,14 @@ module.exports = {
...
+      options.languages.forEach((language) => {
```
Comment (Dhaulagiri) [inline packages/components/index.js:19]: [Example](https://github.com/froala/ember-froala-editor/blob/19fee2c9a2a0ec3a475a504773eb5ac88738fe6f/index.js#L123-L128) of how we could even validate that these languages exist

File: packages/components/index.js [hunk 3]
```diff
@@ -25,5 +31,13 @@ module.exports = {
         'dialog-polyfill-css': 'dialog-polyfill/dist/dialog-polyfill.css',
       },
     },
+    '@embroider/macros': {
```
Comment (Dhaulagiri) [inline packages/components/index.js:34]: [Docs](https://github.com/embroider-build/embroider/blob/main/packages/macros/README.md#setting-configuration-from-an-ember-addon)

File: packages/components/addon/components/hds/code-block/index.js [hunk 4]
```diff
@@ -12,21 +12,19 @@ import Prism from 'prismjs';
 import { setup } from 'prismjs-glimmer';
 import { guidFor } from '@ember/object/internals';
 
-import 'prismjs/plugins/line-numbers/prism-line-numbers';
-import 'prismjs/plugins/line-highlight/prism-line-highlight';
+import { getOwnConfig, importSync } from '@embroider/macros';
```
Comment (Dhaulagiri) [inline packages/components/addon/components/hds/code-block/index.js:15]: [Docs](https://github.com/embroider-build/embroider/blob/main/packages/macros/README.md#getownconfig-getconfig-and-getglobalconfig)

**Review by alex-ju — COMMENTED** [review]
This looks really good! Thanks for taking the time to look into it. I'm happy with this approach, seems to be flexible enough. We can potentially provide a pre-defined list of languages (I believe doesn't need to be exhaustive or to match all available options in prism) and test in in our consumers' codebases. How would you like to move forward with this? Point this to #1687 and get it ready for a review, or shall I cherry-pick this work and continue in the component branch?

**Comment by Dhaulagiri** [comment]
We can reference this but I'll close it for now as I think we'll go with another approach

**Pull request closed without merging**